### DISCUSSION TITLE
What is the progress bar showing "loading workspace lua_ls" over the lower right corner when opening a file?

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#715] [Q&A] [answered]
The following progress bars and messages are shown over the lower right corner when opening a file in lazyvim. 

It seems that some of the plugins, such as [Nvbuddy ](https://github.com/SmiteshP/nvim-navbuddy) and [symbols-outline](https://github.com/simrat39/symbols-outline.nvim), can only be used after these processes are finished.

![2023-05-06 10 57 50](https://user-images.githubusercontent.com/23435810/236595586-c121d46c-67b2-494b-8781-e9dc7cefe6b6.png)

![2023-05-06 11 01 26](https://user-images.githubusercontent.com/23435810/236595599-4919d08f-55cb-4e17-aaf1-be57c72ed6b3.png)

I'd like to ask:

1. Which plugin(s) is running in the background and is generating these bars and messages?
2. What do these progress bars and messages mean? 

Thanks.

**Comment by amaanq** [comment]
Lsp

**Reply by hopezh** [reply]
I noticed, based on the [OatVim](https://github.com/hopezh/OatVim) config on which I'm working, that these messages are shown after [noice.nvim](https://github.com/folke/noice.nvim) is installed.

**Reply by dpetka2001** [reply, marked as answer]
`noice` only shows these messages to the user as a visual representation. Lsp is the one that generates and provides them.

**Reply by cosmojg** [reply]
Any idea why there are two loading bars for the same workspace?

**Reply by jyuan0** [reply]
> Any idea why there are two loading bars for the same workspace?

One for the `workspace` scope, and one for the `<fallback>` scope: https://github.com/LuaLS/lua-language-server/wiki/FAQ#why-are-there-two-workspacesprogress-bars
